examples/tensorflow/train_predict.py

```diff
@@ -2,53 +2,69 @@
 from __future__ import division
 from __future__ import print_function
 
+import argparse
 import mlflow
 from mlflow import tensorflow, tracking, pyfunc
 import numpy as np
 import pandas as pd
 import shutil
 import tempfile
 import tensorflow as tf
+import os.path
 
+# Command-line arguments
+parser = argparse.ArgumentParser(description='Tensorflow Example')
+# parser.add_argument('--hidden_units', nargs='*', default='50 20'.split( ), metavar='L',
+#                    help='hidden units (default: [50, 20])')  #  https://docs.python.org/3/library/argparse.html#nargs
+parser.add_argument('--hidden_units', type=str, default='50 20', metavar='L',
+                    help='hidden units (default: "50 20")')
+parser.add_argument('--steps', type=int, default=1000, metavar='N',
+                    help='steps (default: 1000)')
+parser.add_argument('--tag_constants', type=str, default=tf.saved_model.tag_constants.SERVING, metavar='S',
+                    help='tf.saved_model.tag_constants (default: tf.saved_model.tag_constants.SERVING)')
+args = parser.parse_args()
+
+if __name__ == "__main__":
+    # The Estimator periodically generates "INFO" logs; make these logs visible.
+    tf.logging.set_verbosity(tf.logging.INFO)
+    #    tf.app.run(main=main)
 
-def main(argv):
-    # Builds, trains and evaluates a tf.estimator. Then, exports it for inference, logs the exported model
-    # with MLflow, and loads the fitted model back as a PyFunc to make predictions.
-    (x_train, y_train), (x_test, y_test) = tf.keras.datasets.boston_housing.load_data()
-    # There are 13 features we are using for inference.
-    feat_cols = [tf.feature_column.numeric_column(key="features", shape=(x_train.shape[1],))]
-    feat_spec = {"features": tf.placeholder("float", name="features", shape=[None, x_train.shape[1]])}
-    hidden_units = [50, 20]
-    steps = 1000
-    regressor = tf.estimator.DNNRegressor(hidden_units=hidden_units, feature_columns=feat_cols)
-    train_input_fn = tf.estimator.inputs.numpy_input_fn({"features": x_train}, y_train, num_epochs=None, shuffle=True)
     with mlflow.start_run() as tracked_run:
-        mlflow.log_param("Hidden Units", hidden_units)
-        mlflow.log_param("Steps", steps)
-        regressor.train(train_input_fn, steps=steps)
+        tf.logging.set_verbosity(tf.logging.INFO)
+        # for key, value in vars(args).items():
+        #    mlflow.log_param(key, value)
+        hidden_units = [ int( s ) for s in args.hidden_units.split( ) ]  #  [50, 20]
+        steps = args.steps  #  1000
+        print( '(hdnNts, stp) = ' , hidden_units , steps )
```

Comment:
Let's include a description of the parameter information that we're logging and log these values with `mlflow.log_param`. Maybe we can replace this line with:

```
print("Hidden Units: {}".format(hidden_units))
print("Training Steps: {}".format(args.steps))
mlflow.log_param("hidden_units", hidden_units)
mlflow.log_param("training_steps", args.steps)
```